app/crud/async_db/vector_tiles/filters.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from fastapi import HTTPException
+from shapely.geometry import box
+from sqlalchemy.sql.elements import TextClause
+
+from ....application import db
+from ....utils.geostore import get_geostore_geometry
+
+Filter = Tuple[TextClause, Dict[str, Any]]
+Bounds = Tuple[float, float, float, float]
+
+
+async def geometry_filter(
+    geostore_id: Optional[str], bounds: Bounds
+) -> Optional[TextClause]:
+    if isinstance(geostore_id, str):
+        geometry, envelope = await get_geostore_geometry(geostore_id)
+        if not envelope.intersects(box(*bounds)):
+            raise HTTPException(
+                status_code=404, detail="Tile does not intersect with geostore"
+            )
+
+        f = filter_intersects("t.geom", geometry)  # TODO avoid having to use t.
+        return f
+    return None
+
+
+def confidence_filter(high_confidence_only: Optional[bool]) -> Optional[TextClause]:
+    if high_confidence_only:
```

Comment:
I would move this to nasa_viirs_fire_alerts.py since it's only related to VIIRS

Reply:
i guess we can even replace it with a generic contextual filter once we made sure that fire data are normalized